### ISSUE TITLE
train_test_split, sample option random

### ISSUE THREAD

**Issue by lepelletieralexandre** [vertica/VerticaPy#254]
Hello,

We are a problem with train_test_split function and sample function with parameter method = random. 

When we use 2 times these functions, we doesn't have the same results. And when we want 30/70 partition we can have 25/75 or 35/65 (or worse). 

I saw the source code and I saw you create a new column random, and take lines where random column < 0.3. But random is random. 

I think you should use quantile function to select your lines like this : random column < quantile(0.3). 

with this code for sample function, I obtain 0.33 of my dataset exactly: 
```
x=0.33
print("true value : ",len(titanic)*x)
titanic = load_titanic(schema = schema,cursor = cur, name='titanic')
titanic.eval("random", "random()")
drop(schema + ".titanic2" ,cur,method = "table")
titanic.to_db(schema + ".titanic2" , relation_type = "table")
titanic2 = vDataFrame(input_relation = schema + ".titanic2", cursor = cur)
len(titanic2.search("random < " + str(titanic2["random"].quantile(x))))
```

the result is : 

![image](https://user-images.githubusercontent.com/49387599/153617044-36b944ba-21e9-4e35-86a9-222d7deb16d3.png)

Can you change this in sample and train_test_split functions? 

Regards,

Alexandre

**Comment by oualib** [comment]
Hi Alexandre,

You can use the parameter "random_state" to seed the randomness.

Badr

**Comment by lepelletieralexandre** [comment]
Hello,

Even if we use the parameter "random_state", if we want test_size = 0.3, the test size varies between 0.25 and 0.35. 

If you use quantile(test_size) instead of test_size, we can have exactly 0.3 for test size. 

Alexandre

**Comment by oualib** [comment]
That's a good improvement, we can seed the randomness and use a quantile to filter. Thanks a lot for your help.
